api/node_identity.go

```diff
@@ -0,0 +1,27 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: MPL-2.0
+
+package api
+
+type NodeIdentityRenewRequest struct {
+	NodeID string
+}
+
+type NodeIdentityRenewResponse struct{}
+
+type NodeIdentity struct {
+	client *Client
+}
+
+func (n *Nodes) Identity() *NodeIdentity {
+	return &NodeIdentity{client: n.client}
+}
+
+func (n *NodeIdentity) Renew(req *NodeIdentityRenewRequest, qo *QueryOptions) (*NodeIdentityRenewResponse, error) {
```

Comment:
Non-blocking: we're using `QueryOptions` in the RPC handler because we need to be able to set the stale flag. But providing it in the HTTP API exposes a bunch of "knobs" like filters, blocking queries, etc. that we don't actually support. Seeing as how this is our public Go API, maybe we want to point that out in the docstring?

Reply:
Good shout; I added some details within https://github.com/hashicorp/nomad/pull/26277/commits/a5999c7350646b97bab6a150e551ff68108ef176